AnalysisNtuple/makeAnalysisNtuple.C

```diff
@@ -702,13 +725,70 @@ makeAnalysisNtuple::makeAnalysisNtuple(int ac, char** av)
 	    }
 	}
 
+	if( isMC && doOverlapInvert_GJ){
+	    //if (!overlapRemoval_Tchannel(tree)){
+	    if (!overlapRemoval(tree, 25., 2.5, 0.4, tree->event_==eventNum)){
+		count_overlap++;
+		continue;
+	    }
+	}
+
+	if( isMC && doOverlapRemoval_QCD){
+	    if (overlapRemoval(tree, 25., 2.5, 0.4, tree->event_==eventNum)){
+		count_overlap++;
+		continue;
+	    }
+	}
+
 	// //		Apply systematics shifts where needed
 	if( isMC ){
 	    if (jecvar012_g != 1){
 		jecvar->applyJEC(tree, jecvar012_g); // 0:down, 1:norm, 2:up
 	    }
 	}
 
+	//HEM test 
+	int nHEM_ele=0;
+	bool HEM_ele_Veto=false;
+    	for(int eleInd = 0; eleInd < tree->nEle_; ++eleInd){
+        	double eta = tree->eleEta_[eleInd];
+ 		double pt = tree->elePt_[eleInd];
+        	double phi = tree->elePhi_[eleInd];
+		bool ele_HEM_pt_pass  = pt >= 15 ;
+        	bool ele_HEM_eta_pass = eta > -3.0 && eta < -1.4 ;
+        	bool ele_HEM_phi_pass = phi > -1.57 && phi < -0.87;
+        	if ( ele_HEM_pt_pass &&  ele_HEM_eta_pass &&  ele_HEM_phi_pass) nHEM_ele++;
+	}
+  	if (nHEM_ele>=1) {
+        	HEM_ele_Veto=true;
+        }
+ 	int nHEM_pho=0;
+    	bool HEM_pho_Veto=false;
+    	for(int phoInd = 0; phoInd < tree->nPho_; ++phoInd){
+        	double et = tree->phoEt_[phoInd];
+        	double eta = tree->phoEta_[phoInd];
+        	double phi = tree->phoPhi_[phoInd];
+ 		bool pho_HEM_eta_pass =  eta > -3.0   && eta < -1.4 ;
+        	bool pho_HEM_et_pass =  et >= 15;
+        	bool pho_HEM_phi_pass = phi > -1.57  && phi < 0.87 ;
+        	if (pho_HEM_eta_pass && pho_HEM_phi_pass && pho_HEM_et_pass) {nHEM_pho++ ;}
+	}
+	if (nHEM_pho>=1) {
+        	HEM_pho_Veto=true;
+        }
+
+
+	if((HEM_pho_Veto || HEM_ele_Veto) && year=="2018") isHemVetoObj =true ;
```

Comment:
isHemVetoObj is not reset to false in each event.  This is a bug, so any event AFTER the first hem veto would also get vetoed, no matter if it is truly supposed to be or not.  The bette way to do this is to set isHemVetoObj to the conditional statement:
`isHemVetoObj = (HEM_pho_Veto || HEM_eleVeto) && year=="2018"`

Reply:
Yes you are right. That was not getting reset to false. Thanks for pointing this out.